src/app/child-dev-project/previous-schools/previous-schools.component.spec.ts

```diff
@@ -76,4 +85,69 @@ describe("PreviousSchoolsComponent", () => {
       done();
     });
   });
+
+  it("should only show columns which are defined by the config", fakeAsync(() => {
+    const config: PanelConfig = {
+      entity: new Child(),
+      config: {
+        single: true,
+        columns: [
+          { id: "schoolId", label: "Team", input: "school" },
+          { id: "start", label: "From", input: "date" },
+          { id: "end", label: "To", input: "date" },
+        ],
+      },
+    };
+    component.onInitFromDynamicConfig(config);
+    component.ngOnInit();
+    tick();
+    expect(component.columns).toHaveSize(3);
+    let columnNames = component.columns.map((column) => column.label);
+    expect(columnNames).toContain("Team");
+    expect(columnNames).toContain("From");
+    expect(columnNames).toContain("To");
+
+    config.config.columns.push({
+      id: "schoolClass",
+      label: "Class",
+      input: "text",
+    });
+    config.config.columns.push({
+      id: "result",
+      label: "Result",
+      input: "percentageResult",
+    });
+
+    component.onInitFromDynamicConfig(config);
+    component.ngOnInit();
+    tick();
+    expect(component.columns).toHaveSize(5);
+    columnNames = component.columns.map((column) => column.label);
+    expect(columnNames).toContain("Team");
+    expect(columnNames).toContain("From");
+    expect(columnNames).toContain("To");
+    expect(columnNames).toContain("Class");
+    expect(columnNames).toContain("Result");
+  }));
+
+  it("should display errors for invalid fields", () => {
+    const entry = new ChildSchoolRelation();
+    let validation = component.formValidation(entry);
+    expect(validation.hasPassedValidation).toBeFalse();
+
+    entry.schoolId = "test school";
```

Comment:
if each of these would be a separate test with a title, it would be slightly easier to grasp to intended scenario that is tested